tests/integration/ovs_test.py

```diff
@@ -387,6 +387,102 @@ def test_ovs_vlan_access_tag():
     assertlib.assert_absent(PORT1)
 
 
+@pytest.mark.tier1
+@pytest.mark.skipif(
+    nm_major_minor_version() < 1.41,
+    reason="OVS VLAN trunks was not supported in NM",
+)
+def test_ovs_vlan_trunks():
+    bridge = Bridge(BRIDGE1)
+    bridge.add_internal_port(PORT1, ipv4_state={InterfaceIPv4.ENABLED: False})
+    bridge.set_port_option(
+        PORT1,
+        {
+            OVSBridge.Port.NAME: PORT1,
+            OVSBridge.Port.VLAN_SUBTREE: {
+                OVSBridge.Port.Vlan.MODE: OVSBridge.Port.Vlan.Mode.TRUNK,
+                OVSBridge.Port.Vlan.TRUNK_TAGS: [
+                    {OVSBridge.Port.Vlan.TrunkTags.ID: 10},
+                    {OVSBridge.Port.Vlan.TrunkTags.ID: 20},
+                    {
+                        OVSBridge.Port.Vlan.TrunkTags.ID_RANGE: {
+                            OVSBridge.Port.Vlan.TrunkTags.MIN_RANGE: 30,
+                            OVSBridge.Port.Vlan.TrunkTags.MAX_RANGE: 40,
+                        }
+                    },
+                ],
+            },
+        },
+    )
+
+    with bridge.create() as state:
+        assertlib.assert_state_match(state)
+
+    assertlib.assert_absent(BRIDGE1)
+    assertlib.assert_absent(PORT1)
+
+
+@pytest.mark.tier1
+@pytest.mark.skipif(
+    nm_major_minor_version() < 1.41,
+    reason="OVS VLAN trunks was not supported in NM",
+)
+def test_remove_ovs_vlan_trunks():
+    bridge = Bridge(BRIDGE1)
+    bridge.add_internal_port(PORT1, ipv4_state={InterfaceIPv4.ENABLED: False})
+    bridge.set_port_option(
+        PORT1,
+        {
+            OVSBridge.Port.NAME: PORT1,
+            OVSBridge.Port.VLAN_SUBTREE: {
+                OVSBridge.Port.Vlan.MODE: OVSBridge.Port.Vlan.Mode.TRUNK,
+                OVSBridge.Port.Vlan.TRUNK_TAGS: [
+                    {OVSBridge.Port.Vlan.TrunkTags.ID: 100},
+                    {OVSBridge.Port.Vlan.TrunkTags.ID: 300},
+                ],
+            },
+        },
+    )
+
+    desired_state = bridge.state
+    libnmstate.apply(desired_state)
```

Comment:
Before this line, they should be stored as `pytest.fixture`, so that any error happening in the test case will trigger clean up in this fixture. Your current code will have leftover if any failure happens.

Reply:
Done